qadence_protocols/mitigations/readout.py

```diff
@@ -262,13 +273,15 @@ def mitigation_minimization(
 def mitigate(
     model: QuantumModel,
     options: dict,
-    noise: Noise | None = None,
+    noise: NoiseHandler | None = None,
     param_values: dict[str, Tensor] = dict(),
 ) -> list[Counter]:
-    if noise is None or noise.protocol != Noise.READOUT:
-        if model._noise is None or model._noise.protocol != Noise.READOUT:
+    if noise is None or (not isinstance(noise.protocol[-1], NoiseProtocol.READOUT)):
```

Comment:
Do you have to use `isinstance` here ? if `noise.protocol` is expected to be a list of `NoiseProtocol` it shouldn't be a problem to compare them ?

Reply:
Using filter now